addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -313,9 +315,10 @@ static String getRefreshUrl(String value) {
     }
 
     private static final Pattern LOCATION_PATTERN =
-            Pattern.compile("(?i)\\s*url\\s*=\\s*[\"']?([^'\"]*)[\"']?");
+            Pattern.compile("(?i)^\\s*url\\s*=\\s*[\"']?([^'\"]*)[\"']?");
 
     static String getLocationUrl(String value) {
+        value = value.replaceFirst("\\d?;", "");
```

Comment:
I need to look into this bit further. I was needed to knee tests compliant but now that I’m mulling it over I’m not sure it’s correct.

Also, I realized the first bit of the PR/commit message is now wrong, so I’ll tweak that later too.